src/routes/muzzle-route.ts

```diff
@@ -1,43 +1,52 @@
-import { WebClient } from "@slack/web-api";
+import {
+  ChatDeleteArguments,
+  ChatPostMessageArguments,
+  WebClient
+} from "@slack/web-api";
 import express, { Request, Response, Router } from "express";
 import {
-  IDeleteMessageRequest,
   IEventRequest,
-  IPostMessageRequest,
   ISlashCommandRequest
 } from "../shared/models/slack/slack-models";
 import {
   addUserToMuzzled,
-  isMuzzled,
-  muzzle
+  muzzle,
+  muzzled
 } from "../utils/muzzle/muzzle-utils";
 import { getUserId, getUserName } from "../utils/slack/slack-utils";
 
 export const muzzleRoutes: Router = express.Router();
 const muzzleToken: any = process.env.muzzleBotToken;
 const web: WebClient = new WebClient(muzzleToken);
+const MAX_SUPPRESSIONS: number = 7;
 
 muzzleRoutes.post("/muzzle/handle", (req: Request, res: Response) => {
   const request: IEventRequest = req.body;
   console.log(request);
-  if (isMuzzled(request.event.user)) {
+  if (muzzled.has(request.event.user)) {
     console.log(`${request.event.user} is muzzled! Suppressing his voice...`);
-    const deleteRequest: IDeleteMessageRequest = {
+    const deleteRequest: ChatDeleteArguments = {
       token: muzzleToken,
       channel: request.event.channel,
       ts: request.event.ts,
       as_user: true
     };
 
-    const postRequest: IPostMessageRequest = {
-      token: muzzleToken,
-      channel: request.event.channel,
-      text: `<@${request.event.user}> says "${muzzle(request.event.text)}"`
-    };
-
     web.chat.delete(deleteRequest).catch(e => console.error(e));
 
-    web.chat.postMessage(postRequest).catch(e => console.error(e));
+    if (muzzled.get(request.event.user)!.suppressionCount < MAX_SUPPRESSIONS) {
```

Comment:
I might just be a dumbass but what exactly is this funky syntax doing muzzled.get(request.event.user)! 👀

Reply:
Essentially, it is a guard that ensure that the result of the get is not `undefined`. Without this, TS throw an error stating that the result of the `get` could be undefined... Which is kind of frustrating because we explicitly check that this key exists in the map on line 26.

Reply:
Maybe lodash get would be a little easier to read? 🤷‍♂

Reply:
`get(muzzled.get(request.event.user), 'suppressionCount', 0)`